File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    heapq.heapify(scoville)
    
    while len(scoville)>1:
        first_min = heapq.heappop(scoville)
        if first_min >= K:
            return answer
        second_min = heapq.heappop(scoville)
        new_score = first_min + second_min*2
        heapq.heappush(scoville, new_score)
        answer+=1
    
    num = scoville.pop()
    if num >=K: return answer
    else: return -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

int solution(vector<int> scoville, int K) {
    int answer = 0;
    priority_queue<int, vector<int>, greater<int>> pq;
    for(auto a: scoville) pq.push(a);
    int cnt=0;
    while(!pq.empty()){
        if(pq.top()<K){
            int b = pq.top();
            pq.pop();
            
            if(pq.empty()) return -1;
            
            int c = pq.top();
            pq.pop();
            pq.push(b+c*2);
            cnt++;

        }
        else{
            return cnt;
        }
    }
}